tools/apprunner-common/src/main/java/org/projectnessie/quarkus/runner/ProcessHandler.java

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.quarkus.runner;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.function.LongSupplier;
+
+/**
+ * Handles the execution of an external process, focused on running a Quarkus application jar.
+ *
+ * <p>Starts the process configured in a {@link ProcessBuilder}, provides a method to get the {@link
+ * #getListenUrl() Quarkus HTTP listen URL} as Quarkus prints to stdout, and manages process
+ * lifetime and line-by-line I/O pass-through for stdout + stderr.
+ *
+ * <p>Any instance of this class can only be used to start (and stop) one process and cannot be
+ * reused for another process.
+ *
+ * <p>This implementation is not thread-safe.
+ */
+public class ProcessHandler {
+
+  // intentionally long timeouts - think: slow CI systems
+  public static final long MILLIS_TO_HTTP_PORT = 30_000L;
+  public static final long MILLIS_TO_STOP = 15_000L;
+
+  private LongSupplier clock = System::nanoTime;
```

Comment:
I still don't like calling this a clock. Its a ticker or counter

Reply:
renamed to `ticker`